### ISSUE TITLE
che-7.17.0 Failing to come up with Default TLS - Exception while retrieving OpenId configuration from endpoint

### ISSUE THREAD

**Issue by chaitu454** [eclipse-che/che#17650]
### Describe the bug
che-7.17.0 Failing to come up with Default TLS

### Che version

 - [ ] latest
 - [ ] nightly
 - [x] other: 7.17.0

### Steps to reproduce

1) Spin Chesever7.17.0 using chectl 7.17.0 with below command 
chectl server:start --platform=docker-desktop
2) Now stop and delete this server. Now spin up with -self-signed certificate. I know the option is deprecated but it will start searching for the selfsigner and try to come up.
chectl server:start --platform=docker-destkop --self-signed-cert
3) Now if you bring down the server and try to bring it up in either TLS mode or Self-signed it will fail in both the cases.
4) Incase of TLS it throws the below Keycloak error and incase of Self-signed it says che-certificate-manager already exists even when the kubernetes cluster is cleaned up that's by deleting all PVs, PVCs, PODs and other resources.

### Expected behavior
Given this is docker container based ideally no change tried in earlier run should be persisted unless asked for and Che server with 7.17.0 image should come up with other pods like key cloak and all.

### Runtime
   - [ ] kubernetes (include output of `kubectl version`)
   - [ ] Openshift (include output of `oc version`)
   - [ ] minikube (include output of `minikube version` and `kubectl version`)
   - [ ] minishift (include output of `minishift version` and `oc version`)
   - [x] docker-desktop + K8S (include output of `docker version` and `kubectl version`) 
   - [ ] other: (please specify)

 **kubectl version**
Client Version: version.Info{Major:"1", Minor:"16+", GitVersion:"v1.16.6-beta.0", GitCommit:"e7f962ba86f4ce7033828210ca3556393c377bcc", GitTreeState:"clean", BuildDate:"2020-01-15T08:26:26Z", GoVersion:"go1.13.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"16+", GitVersion:"v1.16.6-beta.0", GitCommit:"e7f962ba86f4ce7033828210ca3556393c377bcc", GitTreeState:"clean", BuildDate:"2020-01-15T08:18:29Z", GoVersion:"go1.13.5", Compiler:"gc", Platform:"linux/amd64"}

**docker version**

Client: Docker Engine - Community
 Version:           19.03.12
 API version:       1.40
 Go version:        go1.13.10
 Git commit:        48a66213fe
 Built:             Mon Jun 22 15:41:33 2020
 OS/Arch:           darwin/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.12
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.13.10
  Git commit:       48a66213fe
  Built:            Mon Jun 22 15:49:27 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.2.13
  GitCommit:        7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc:
  Version:          1.0.0-rc10
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683

### Installation method
   - [x] chectl
     * provide a full command that was used to deploy Eclipse Che (including the output)
     * provide an output of `chectl version` command
   - [ ] OperatorHub
   - [ ] I don't know

**./chectl version**
chectl/7.17.0 darwin-x64 node-v10.22.0

./chectl server:start --platform=docker-desktop
**Output-** 
 ❯ Eclipse Che pod bootstrap
      ✔ scheduling...done.
      ✔ downloading images...done.
      ✖ starting
        → ERR_TIMEOUT: Timeout set to pod ready timeout 130000
      Retrieving Eclipse Che server URL
      Eclipse Che status check
    Retrieving Keycloak admin credentials

Error --
====================================================================================
1) Error injecting constructor, java.lang.RuntimeException: Exception while retrieving OpenId configuration from endpoint: https://keycloak-che.192.168.1.2.nip.io/auth/realms/che/.well-known/openid-configuration
  at org.eclipse.che.multiuser.keycloak.server.KeycloakSettings.<init>(KeycloakSettings.java:71)
  at org.eclipse.che.multiuser.keycloak.server.KeycloakSettings.class(KeycloakSettings.java:54)
  while locating org.eclipse.che.multiuser.keycloak.server.KeycloakSettings
    for the 1st parameter of org.eclipse.che.multiuser.keycloak.server.KeycloakProfileRetriever.<init>(KeycloakProfileRetriever.java:40)
  at org.eclipse.che.multiuser.keycloak.server.KeycloakProfileRetriever.class(KeycloakProfileRetriever.java:33)
  while locating org.eclipse.che.multiuser.keycloak.server.KeycloakProfileRetriever
    for the 1st parameter of org.eclipse.che.multiuser.keycloak.server.dao.KeycloakProfileDao.<init>(KeycloakProfileDao.java:38)
  while locating org.eclipse.che.multiuser.keycloak.server.dao.KeycloakProfileDao
=======================================================================================
18-Aug-2020 09:28:08.039 INFO [Annotated-scheduler-0] org.apache.catalina.loader.WebappClassLoaderBase.checkStateForResourceLoading Illegal access: this web application instance has been stopped already. Could not load [io.micrometer.core.instrument.util.TimeUtils]. The following stack trace is thrown for debugging purposes as well as to attempt to terminate the thread which caused the illegal access.
	java.lang.IllegalStateException: Illegal access: this web application instance has been stopped already. Could not load [io.micrometer.core.instrument.util.TimeUtils]. The following stack trace is thrown for debugging purposes as well as to attempt to terminate the thread which caused the illegal access.
		at org.apache.catalina.loader.WebappClassLoaderBase.checkStateForResourceLoading(WebappClassLoaderBase.java:1378)
========================================================================================

### Environment
   - [x] my computer
       - [ ] Windows
       - [ ] Linux
       - [x] macOS
   - [ ] Cloud
       - [ ] Amazon
       - [ ] Azure
       - [ ] GCE
       - [ ] other (please specify)
   - [ ] other: please specify

## Summary
Unable to spinup che server - 7.17.0 with default TLS or Self signed certs. Keycloak error.

## Relevant information

I am trying to spin up che server 7.17.0 using chectl on docker-desktop. I am using this setup since some months but I never faced this issue earlier. 

I did a custom che server build and was trying to test it locally. It was working till Saturday then suddenly I started getting this error first I thought it has something to do with my customisation but as is I didn't touch anything related to Keycloak in my customisation. So I stopped passing --cheimage parameter to chectl and tried spinning up the default image it brings in.

Even there I started seeing this issue. I have tried completely cleaning up all the namespaces in my cluster  restarted kubernetes multiple times but no use. So I thought of switching off TLS just incase it's causing the issue and tried with Self-signed certificate but it started failing with error as Che Certificate manager is already in place.

So in short now its failing with default TLS as well as self signed certificate even with original che 7.17.0 image as well. Infact in suspicision I have downloaded chectl 7.17.0 again but no use.

I am completely stuck with this issue and any suggestion progress will be very useful. Attached is the error log.

[Che_error.log](https://github.com/eclipse/che/files/5089376/Che_error.log)

**Comment by tolusha** [comment]
This is kind of bug. Pls provide additional information by tagging items in the description.

**Comment by chaitu454** [comment]
@tolusha Please let me know incase any further details needed to fix this issue. Is there any thing I can try to resume my local che server while this is being investigated. Main problem is I am out of options to try ...

**Comment by chaitu454** [comment]
In fact I am running this same cheserver7.17.0 on GCP and its working fine there. There we are using Keycloak 7.12.0 and did that deployment through helm directly. This chectl setup was running till saturday before I tried above steps

**Comment by tolusha** [comment]
> chectl server:start --platform=docker-desktop

When you spin up the Che Server the very first time, did it work?
And how did you stop and delete the server?

**Comment by chaitu454** [comment]
chectl server:stop
chectl server:delete

**Comment by tolusha** [comment]
@chaitu454 
I still have one question required the issue:
> When you spin up the Che Server the very first time, did it work?

**Comment by chaitu454** [comment]
@tolusha  Yes it did. Since I tried to switch between self-signed and tls option it broke

**Comment by tolusha** [comment]
Could you remove the namespace and redeploy Eclipse Che again?

**Comment by chaitu454** [comment]
Actually, since I reported the issue I have reset docker-desktop and it worked. In between when I ran out of options I have tried with --installer helm

**Comment by chaitu454** [comment]
Further observation on this is when I try to do chectl server:stop or delete it is asking for Authorisation token.
As per https://www.eclipse.org/che/docs/che-7/authenticating-users/#authenticating-to-the-che-server-using-openid_authenticating-to-the-che-server if I try and got to https://che-che.192.168.1.2.nip.io/api/keycloak/settings and then to get token endpoint it's throwing **internal server exception**. 

Hence initial deletes are not successful sometimes which seems to be the reason for above redundancy in components. When Chectl is trying to create new components and trying to update it's unable to do so as Keycloak token is not active anymore. So if I clear the complete Namespace or Reset Docker-Desktop then it's getting resolved. 

I am not sure if this theory is right. Can be complete hoax . Please suggest.
